ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -550,26 +592,20 @@ private boolean isTtlUpdateEntryValid(StoreKey key, IndexValue ttlUpdateValue, l
       valid = false;
     } else if (putValue.getOffset().getName().equals(ttlUpdateValue.getOffset().getName())) {
       // can be invalid if it is in the same log segment and the put is invalid
-      valid = isPutEntryValid(key, putValue, referenceTimeInMs, deletedKeys);
+      IndexFinalState finalState = keyFinalStates.get(key);
+      if (finalState != null && finalState.isDelete() && finalState.getOperationTime() < referenceTimeInMs) {
```

Comment:
why this is not `finalState.getOperationTime() > referenceTimeInMs`?

Reply:
finalState is delete so the operation time is when this blob is deleted. when deletedAt is less than reference time, it means this PUT value is not valid. For example. the reference is now - 7days and the deletedAt is now - 10 days, it means the blob is deleted 10 days ago and we only have 7 days as retention duration. now -10 days < now - 7 days, the PUT is invalid, so as the TTL_UPDATE.

Reply:
I see, make sense.